src/ppx/graphics_util.cpp

```diff
@@ -775,6 +782,68 @@ Result CreateImageFromFile(
 
 // -------------------------------------------------------------------------------------------------
 
+Result CreateImageFromArray(
+    grfx::Queue*                 pQueue,
+    grfx::Image**                ppImage,
+    const ImageOptions&          options,
+    float*                       array,
+    const int                    imageWidth,
+    const int                    imageHeight)
+{
+    //TODO(ghanz): generalize float* with void*
+    PPX_ASSERT_NULL_ARG(pQueue);
+    PPX_ASSERT_NULL_ARG(ppImage);
+
+    Timer timer;
+    PPX_ASSERT_MSG(timer.Start() == ppx::TIMER_RESULT_SUCCESS, "timer start failed");
+    double fnStartTime = timer.SecondsSinceStart();
+
+    Result ppxres;
+    // Create bitmap from array
+    Bitmap bitmap;
+    Bitmap* pBitmap = &bitmap;
+    Bitmap::Format format = Bitmap::FORMAT_RGBA_FLOAT;
+    ppxres = Bitmap::Create(static_cast<uint32_t>(imageWidth), static_cast<uint32_t>(imageHeight), format, pBitmap);
+    size_t nbytes = Bitmap::StorageFootprint(static_cast<uint32_t>(imageWidth), static_cast<uint32_t>(imageHeight), format); 
+
+    // Fill array data to bitmap at Red channel
+    char* pSrc = (char *)array;
+    char* pDst  = pBitmap->GetData();
+    const uint32_t srcChannelSize = sizeof(float);
+    const uint32_t dstPixelStride = pBitmap->GetPixelStride();
+    for (uint32_t y = 0; y < pBitmap->GetHeight(); ++y) {
+        for (uint32_t x = 0; x < pBitmap->GetWidth(); ++x) {
+            memcpy(pDst, pSrc, srcChannelSize);
+            pSrc += srcChannelSize;
+            pDst += dstPixelStride;
+        }
+    }
+    free(array);
```

Comment:
I think usually the caller frees the data? Or make it clear in the function comment

Reply:
The function Bitmap::LoadFile() from bitmap.cpp creates a pointer float* pData, then free it at row #624  `stbi_image_free(pData);`, should I be doing the same?

Reply:
As Jun mentioned Allocating a memory in one place and deallocating elsewhere is a Big no.